regclient/regclient
# Refactoring regclient API

### Fixes issue

n/a


### Describe the change

This is a significant refactor to move the registry specific API's into their own location, allowing other schemes, like OCI layout support, to be added later. This also removes some of stuttering in various import paths with the extra `regclient` directory.

Breaking: most of the old names have aliases to allow existing code to work unmodified. One exception is `RepoListWithOpts` which is now only `RepoList` using function opts instead of the previous struct of opts.




### How to verify it

Most code should build unchanged because of aliases setup to point to new locations. To force new import paths, building with `-tags nolegacy` will disable the various aliases which will eventually be removed. The http and retry code have been significantly redesigned and deserves extra testing.


### Changelog text

Refactoring API to allow other scheme types.


### Please verify and check that the pull request fulfills the following requirements

Documentation for end user tools should be unchanged with the refactor. Because of the scope of changes, this was not squashed to a single commit, but changes to common areas have been squashed resulting in some of these commits appearing out of order.
